adapters/adnuntius/adnuntius.go

```diff
@@ -369,6 +375,38 @@ func getGDPR(request *openrtb2.BidRequest) (string, string, error) {
 	return gdpr, consent, nil
 }
 
+func generateReturnExt(ad Ad, request *openrtb2.BidRequest) (json.RawMessage, error) {
+	// We always force the publisher to render
+	var adRender int8 = 0
+
+	var requestRegsExt *openrtb_ext.ExtRegs
+	if request.Regs != nil && request.Regs.Ext != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &requestRegsExt); err != nil {
+
+			return nil, fmt.Errorf("Failed to parse Ext information in Adnuntius: %v", err)
+		}
+	}
+
+	if requestRegsExt != nil && requestRegsExt.DSA != nil && requestRegsExt.DSA.PubRender != &adRender {
```

Comment:
The comparison `requestRegsExt.DSA.PubRender != &adRender` will always be not equal since you're comparing memory addresses and the memory address of `requestRegsExt.DSA.PubRender` will never be the same as `adRender`.

I suggest adding another JSON test called `check-dsa-pubrender-zero.json` and renaming your test `check-dsa.json` to `check-dsa-pubrender-non-zero.json`.

Then I think you want to change this conditional to the following so that you are comparing the values instead of the memory addresses:
`if requestRegsExt != nil && requestRegsExt.DSA != nil && requestRegsExt.DSA.PubRender != nil && *requestRegsExt.DSA.PubRender != adRender`

Also, what happens if `regs.ext.dsa.pubrender` is nil or if `regs.ext.dsa.pubrender` is 0? You don't want the dsa object included in the bid response in these cases?